packages/babel-plugin-proposal-decorators/test/fixtures/2023-11-classes/replacement-static-private-environment-super/exec.js

```diff
@@ -0,0 +1,38 @@
+{
+  "different private/super access in a static private method";
+
+  const dummy = () => {}
+
+  let b;
+
+  const idFactory = (hint) => {
+    return class { static ["id" + hint](v) { return v } }
+  }
+
+  class C extends idFactory("C") { #x = "C#x"; }
+
+  const dec = (b) => {
+    originalB = b;
+    return C;
+  }
+
+  class A extends idFactory("A") {
+    #x = "A#x";
+    static *[Symbol.iterator]() {
+      @dec
+      class B extends idFactory("B") {
+        #x = "B#x";
+        @(yield super.idA(o => o.#x), dummy)
+        static *#iter() {
+          yield super.idC(o => o.#x);
+        }
+        static *iter() {
+          yield* super.idB(C.#iter());
+        }
+      }
+      b = new originalB();
+      yield* originalB.iter();
+    }
+  }
+  expect([...A].map(fn => fn(b)).join()).toBe("B#x,B#x");
+}
```

Comment:
Hang tight. This test can be very confusing, I have to carefully specify the receiver to make it a valid JavaScript. I think it is worthy of a compiler assumption such as `DecoratorDoesNotReplaceClass`, to avoid the inner class completely, because once a class is replaced in a class decorator, the common `this` element access could be easily broken.